torch/_subclasses/fake_tensor.py

```diff
@@ -1780,14 +1781,17 @@ def can_run_unsafe_fallback(self, func: OpOverload):
     # fake tensor is not supported.  What we actually wanted to happen
     # was to give the subclass a chance to figure out what it wants to
     # before erroring out. Returning NotImplemented here allows this.
-    def check_for_subclass(self, flat_args):
+    def check_user_dispatchable_args(self, flat_args):
+        # Avoid importing sympy at a module level
+        from torch.fx.experimental.symbolic_shapes import is_singleton
+
         def check(x):
             return (
                 isinstance(x, torch.Tensor)
                 and not isinstance(x, FakeTensor)
                 and type(x) is not torch.Tensor
                 and type(x) is not torch.nn.Parameter
-            )
+            ) or is_singleton(x)
```

Comment:
Does this actually work? If the singleton is inside a list/tuple you won't catch it here right?

Reply:
It's doing this on the tree_flattened args!